crates/ruff/src/autofix/mod.rs

```diff
@@ -59,17 +59,21 @@ fn apply_fixes<'a>(
         })
         .sorted_by(|(rule1, fix1), (rule2, fix2)| cmp_fix(*rule1, *rule2, fix1, fix2))
     {
-        // If we already applied an identical fix as part of another correction, skip
-        // any re-application.
-        if fix.edits().iter().all(|edit| applied.contains(edit)) {
+        // Remove any edits that were applied as part of a previous fix.
+        let mut edits = fix.edits();
+        edits.retain(|edit| !applied.contains(edit));
```

Comment:
Do you think it's feasible to create a `Vec<&Edit>` which needs to be applied? i.e., filtering out the `Edit` which has been applied but not in place. This would avoid cloning the `Edit`.